### PR TITLE
Opportunity: Add Deliver Form UI, Models and saving

### PR DESCRIPTION
**Pull request by pxwxnvermx**
This PR includes code to Load deliver forms based on the Deliver app a user selects and UI to Add a Deliver form to the opportunity.

 
image

### CODE REVIEW

File: commcare_connect/templates/opportunity/opportunity_create.html [hunk 1]
```diff
@@ -12,5 +12,28 @@
       {% crispy form %}
     </form>
 
+    <script>
+      window.onload = function () {
+        const applications = {{ applications|safe }};
```
Comment (snopoke) [inline commcare_connect/templates/opportunity/opportunity_create.html:17]: it's probably be better to use the `json_script` tag to convert the python dict to json safely : https://docs.djangoproject.com/en/4.2/ref/templates/builtins/#std-templatefilter-json_script

File: commcare_connect/templates/opportunity/opportunity_create.html [hunk 2]
```diff
@@ -12,5 +12,28 @@
...
+
+          for (const form of selectedApp.forms) {
+            const form_option = document.createElement("option");
+            form_option.text = form.name;
+            form_option.value = form.id;
+            deliver_form_select.add(form_option);
+          }
+        }
+      }
+    </script>
```
Comment (snopoke) [inline commcare_connect/templates/opportunity/opportunity_create.html:37]: This code is fine - just wondering if you looked at using alpinejs? e.g. https://www.raymondcamden.com/2022/07/29/building-related-selects-in-alpinejs

**Review by snopoke — APPROVED** [review]

File: commcare_connect/templates/opportunity/opportunity_create.html (hunk 2 above)
Comment (pxwxnvermx) [reply commcare_connect/templates/opportunity/opportunity_create.html:37]: I took a look at alpineJS, I was already done with these changes by then.

File: commcare_connect/templates/opportunity/opportunity_create.html (hunk 1 above)
Comment (pxwxnvermx) [reply commcare_connect/templates/opportunity/opportunity_create.html]: Thanks. I included this in the code. [05334bf](https://github.com/dimagi/commcare-connect/pull/37/commits/05334bf5e24d575d2c1a3e711ebd7b2edc5dcd65)

**Pull request merged**